sql/core/src/main/scala/org/apache/spark/sql/execution/AliasAwareOutputPartitioningAndOrdering.scala

```diff
@@ -16,16 +16,18 @@
  */
 package org.apache.spark.sql.execution
 
-import org.apache.spark.sql.catalyst.expressions.{Alias, Attribute, AttributeReference, Expression, NamedExpression}
+import org.apache.spark.sql.catalyst.expressions.{Alias, Attribute, AttributeReference, Expression, NamedExpression, SortOrder}
 import org.apache.spark.sql.catalyst.plans.physical.{HashPartitioning, Partitioning}
 
 /**
- * A trait that handles aliases in the `outputExpressions` to produce `outputPartitioning`
- * that satisfies output distribution requirements.
+ * A trait that handles aliases in the `outputExpressions` and `orderingExpressions` to produce
+ * `outputPartitioning` and `outputOrdering` that satisfy distribution and ordering requirements.
  */
-trait AliasAwareOutputPartitioning extends UnaryExecNode {
+trait AliasAwareOutputPartitioningAndOrdering extends UnaryExecNode {
   protected def outputExpressions: Seq[NamedExpression]
 
+  protected def orderingExpressions: Seq[SortOrder] = child.outputOrdering
```

Comment:
this implicitly indicates that the plan inherits output ordering from its child. This seems risky to me as `SparkPlan.outputOrdering` is `Nil` by default.

Reply:
Good catch. Now that you mention it, should I just separate this into `AliasAwareOutputPartitioning` and `AliasAwareOutputOrdering` and apply the latter only to `SortAggregateExec` and `ProjectExec` so that the intention is clear?

Reply:
Ah, nice catch... Yea, could you try to split it into the two parts?